netstandard/tools/Trimming/FileNode.cs

```diff
@@ -0,0 +1,153 @@
+            }
+
+            if (!string.IsNullOrEmpty(PackageId))
+            {
+                NuGetPackageNode package;
+
+                if (!allPackages.TryGetValue(PackageId, out package))
+                {
+                    // some file came from a package that wasn't found in the lock file
+                }
+                else
+                {
+                    Package = package;
+                    Package.Files.Add(this);
+                }
+            }
+        }
+
+        public bool IsIncluded { get; set; }
+        public string Name { get; }
+        public ITaskItem OriginalItem { get; }
+        public string PackageId { get; }
+        public string SourceFile { get; }
+        public NuGetPackageNode Package { get; }
+        public IEnumerable<FileNode> Dependencies { get { return _dependencies; } }
+
+        public void PopulateDependencies(Dictionary<string, FileNode> allFiles, bool preferNativeImage)
+        {
+            List<FileNode> dependencies = new List<FileNode>();
+
+            try
+            {
+                using (var peReader = new PEReader(new FileStream(SourceFile, FileMode.Open, FileAccess.Read, FileShare.Delete | FileShare.Read)))
+                {
+                    if (peReader.HasMetadata)
+                    {
+                        var reader = peReader.GetMetadataReader();
+                        foreach (var handle in reader.AssemblyReferences)
+                        {
+                            var reference = reader.GetAssemblyReference(handle);
+                            var referenceName = reader.GetString(reference.Name);
+
+                            var referenceCandidates = preferNativeImage ? 
+                                new[] { referenceName + ".ni.dll", referenceName + ".dll" } :
+                                new[] { referenceName + ".dll", referenceName + ".ni.dll" };
+
+                            FileNode referencedFile = null;
+                            foreach (var referenceCandidate in referenceCandidates)
+                            {
+                                if (allFiles.TryGetValue(referenceCandidate, out referencedFile))
+                                {
+                                    break;
+                                }
+                            }
+
+                            if (referencedFile != null)
+                            {
+                                dependencies.Add(referencedFile);
+                            }
+                            else
+                            {
+                                // static dependency that wasn't satisfied, this can happen if folks use 
+                                // lightup code to gaurd the static dependency.
+                                // this can also happen when referencing a package that isn't implemented
+                                // on this platform but don't fail the build here
```

Comment:
I'll consider it.  I'd need to plumb the log down as an abstraction.  I don't want this thing knowing about MSBuild-types.